sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/api/Azure.Iot.TimeSeriesInsights.netstandard2.0.cs

```diff
@@ -329,13 +362,44 @@ public partial class TimeSeriesIdProperty
         public string Name { get { throw null; } }
         public Azure.Iot.TimeSeriesInsights.Models.TimeSeriesIdPropertyTypes? Type { get { throw null; } }
     }
+    public partial class TimeSeriesId<T1> : Azure.Iot.TimeSeriesInsights.ITimeSeriesId
+    {
+        public TimeSeriesId(T1 timeSeriesIdProp1) { }
```

Comment:
I'd like to improve the names of these parameters a bit.

For inspiration, I went to the portal because that is end user-facing. There you see this:
![image](https://user-images.githubusercontent.com/14119884/110009434-dcd6b600-7cd1-11eb-9fa9-452f79d95c99.png)

With a link to <https://docs.microsoft.com/en-us/azure/time-series-insights/how-to-select-tsid?WT.mc_id=Portal-fx>

They seem to call them keys, and when combined call them composite keys.

So I was thinking either `key1` or `compositeKey1`. Thoughts?

Reply:
I see the confusion but on the left to this text box in the image you shared, it is "Property name". 

I do see value in adding description in headers that this property can be a unique key or part of composite key defining a Time Series Id.

Reference: https://docs.microsoft.com/en-us/azure/time-series-insights/how-to-select-tsid

Reply:
The property is what comes from your data source. That is the **external** name. Internally they call them keys.

Reply:
After taking a closer look at the documentation, I tend to agree with David on this one. Property is referred to as the property name coming in from the payload. Where key makes more sense since this can be thought of as a partition key, where you look for entities using a key.